### PR TITLE
35-m3-save-logs-in-database: creatted log function to log and save lo…

### LINKED ISSUE #35: M3: Save logs in database.
All the logs made by the analyzer must be saved in the database.

AC
1. Save the logs.

### PR DESCRIPTION
**Pull request by mcamardon**
…gs and errors in elastic.

### CODE REVIEW

File: src/controllers/video-test.controller.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-        const {headers, data} = await cmcdExtractorService(req, reqURI, {}, dateStart);
-        res.header(headers)
+        const {headers, data} = await cmcdExtractorService(id, req, reqURI, {}, dateStart);
+        res.header(headers);
         data.pipe(res);
     } catch (error) {
         if (error.response) {
             // The client was given an error response (5xx, 4xx)
-            jsLogger.error(error.response.status, error.response.statusText);
+            log(id, {'error.response.status': error.response.status, 'error.response.statusText': error.response.statusText}, error);
```
Comment (JoaquinBCh) [inline src/controllers/video-test.controller.js]: Change the error.response.status for status and error.respons.statusText for statusText

File: src/controllers/video.controller.js [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
         const {concatenatedUrl, decodedJson}  = decodeBase64AndConcat(jsonBase64, videoURL);
-        const {headers, data} = await cmcdExtractorService(req, concatenatedUrl, decodedJson, dateStart);
+        const {headers, data} = await cmcdExtractorService(id, req, concatenatedUrl, decodedJson, dateStart);
         res.header(headers)
         data.pipe(res);
     } catch (error) {
         if (error.response) {
             // The client was given an error response (5xx, 4xx)
-            jsLogger.error(error.response.status, error.response.statusText);
+            log(id, {'error.response.status': error.response.status, 'error.response.statusText': error.response.statusText}, error);
```
Comment (JoaquinBCh) [inline src/controllers/video.controller.js]: Same here

File: src/utils/saveData.js [hunk 3]
```diff
@@ -5,11 +5,12 @@ export const saveData = async (id, validatorRes) => {
     jsLogger.useDefaults({ defaultLevel: jsLogger.TRACE });
 
     try {
+        jsLogger.info(`${id}: Saving data into the database...`);
         await client.index({
-            index: id,
+            index: `logs-${id}`,
```
Comment (JoaquinBCh) [inline src/utils/saveData.js]: The index should be id.

File: src/utils/logger.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import jsLogger from 'js-logger';
+import saveData from './saveData.js';
+
+const log = (id, value, level) => {
+    jsLogger.useDefaults({ defaultLevel: jsLogger.TRACE });
+    jsLogger[level](`${id}: ${JSON.stringify(value)}`);
+    saveData(id, value);
```
Comment (JoaquinBCh) [inline src/utils/logger.js]: the id sended should be `logs-${id}`

**Review by JoaquinBCh — CHANGES_REQUESTED** [review]

**Review by coatesjuan — CHANGES_REQUESTED** [review]
Add execution permissions to the entrypoint.sh file so when we merge the branch it does not break the setup in the docker

**Review by coatesjuan — APPROVED** [review]

**Review by JoaquinBCh — CHANGES_REQUESTED** [review]
We should not log the saving of the logs.

**Review by JoaquinBCh — APPROVED** [review]

**Pull request merged**